### DISCUSSION TITLE
Where does Cyberduck store my SFTP passwords?

### DISCUSSION THREAD

**Discussion by Zen-X** [iterate-ch/cyberduck#15039] [Q&A] [answered]
Hey. I'm a little scared about how Cyberduck behaves after an uninstall. I have:

- deleted the program in the macos program folder
- deleted the file cyberduck.log in ~/Library/Logs/Cyberduck
- deleted the file ~/Library/Preferences/ch.sudo.cyberduck.plist
- deleted the folder Containers/G69SCX94XU.duck/Library/Application Support/duck/

After I reinstall the program and create a new SFTP connection, Cyberduck automatically fills in the correct password after entering the username. And no. I do not have any passwords stored in macos keychain.

How did Cyberduck get my SFTP password? And how can I cleanly delete Cyberduck from my Mac?

Thanks a lot,
Martin

**Comment by tatze96** [comment, marked as answer]
Regarding your questions:
> How did Cyberduck get my SFTP password?

Passwords of bookmarks are generally saved within _Keychain Access.app_ unless you've deactivated the keychain usage within the Cyberduck preferences _Connection_ tab.

The uninstall process leaves the application support folder, settings, and the Keychain entries behind, as they are shared with Mountain Duck. As you've "only" deleted the application support folder, log location, and Cyberduck settings and left the Keychain entries behind, the password will get filled out automatically if you enter the corresponding combination of hostname and username.

> And how can I cleanly delete Cyberduck from my Mac?

To remove Cyberduck completely from your system, clear the following items:
- Remove the application file from the _Applications_ folder.
- Delete the log folder `~/Library/Logs/Cyberduck/`
- Delete the settings file `~/Library/Preferences/ch.sudo.cyberduck.plist`
- Delete the application support folder `~/Library/Group Containers/G69SCX94XU.duck/`
- Clear out all saved login credentials regarding Cyberduck from _Keychain Access.app_.

**Reply by Zen-X** [reply]
Thank you very much. I had not considered the last part with the Keychain Access.app.

Best regards,
Martin
